pallets/utxo/src/lib.rs

```diff
@@ -262,6 +335,48 @@ pub mod pallet {
             }
         }
 
+        /// Create a new stake for the first time.
+        /// This is assumed, that an entity staking its funds also means playing the role of a `Validator`.
+        /// # Arguments
+        /// * `value` - This is assuming that the value is received in a format that's already `x * 1_000, * 1_000_000`
+        /// * `stash_acount` - This is considered as the "bank", which holds the funds. The full meaning is found in `pallet-staking`.
+        /// * `controller_account` = can be considered as the "manager". The full meaning is found in `pallet-staking`.
+        /// * `session_key` - for every new validator candidate, you want to link it to a session key.
+        /// Generation of `session_key` is done through an rpc call `author_rotateKeys`.
+        /// See https://docs.substrate.io/v3/concepts/session-keys/
+        pub fn new_lock_for_staking(
+            value: Value,
+            stash_account: AccountId,
+            controller_account: AccountId,
+            session_key: Vec<u8>,
```

Comment:
Again, same question... why does the user have to provide the session key instead of the software generating it from the controller key, as the docs of substrate say?
https://docs.substrate.io/v3/concepts/session-keys/#:~:text=Substrate%20provides%20the%20Session%20pallet,used%20for%20their%20intended%20purpose.